### DISCUSSION TITLE
Tariff rates different from billing statements

### DISCUSSION THREAD

**Discussion by blacknell** [BottlecapDave/HomeAssistant-OctopusEnergy#1058] [Q&A] [answered]
I'm seeing £0.253578/kWh in the integration vs. £0.24612/kWh on my Octopus statement

Any ideas?

**Comment by bruceskingle** [comment]
Probably VAT @ 5%, not exact for those numbers but if they are derived may be a rounding effect

**Reply by blacknell** [reply]
I don't think so.

The Octopus rate on my statement without VAT is 23.44p which is 24.612p after adding 5% VAT. 25.3578p is much more than this (3% more)

**Comment by blacknell** [comment]
OK. I have managed to figure out that it (this integration) is pulling the wrong tariff from Octopus.

**Comment by blacknell** [comment]
My account has agreement code `E-1R-VAR-22-11-01-H`

When I pull the product `VAR-22-11-01` and look at variant `_H` I get 24.61704p which is the correct rate.

Somehow (and I can't figure it out) it's getting the wrong tariff - possibly not getting the correct variant?

```json
"_H": {
      "varying": {
        "code": "E-1R-VAR-22-11-01-H",
        "standing_charge_exc_vat": 59.2617,
        "standing_charge_inc_vat": 62.224785,
        "online_discount_exc_vat": 0,
        "online_discount_inc_vat": 0,
        "dual_fuel_discount_exc_vat": 0,
        "dual_fuel_discount_inc_vat": 0,
        "exit_fees_exc_vat": 0,
        "exit_fees_inc_vat": 0,
        "exit_fees_type": "NONE",
        "links": [
          {
            "href": "https://api.octopus.energy/v1/products/VAR-22-11-01/electricity-tariffs/E-1R-VAR-22-11-01-H/standing-charges/",
            "method": "GET",
            "rel": "standing_charges"
          },
          {
            "href": "https://api.octopus.energy/v1/products/VAR-22-11-01/electricity-tariffs/E-1R-VAR-22-11-01-H/standard-unit-rates/",
            "method": "GET",
            "rel": "standard_unit_rates"
          }
        ],
        "standard_unit_rate_exc_vat": 23.4448,
        "standard_unit_rate_inc_vat": 24.61704
      }
    },
```

**Reply by BottlecapDave** [reply]
Hello. The tariff is based on the data provided by OE APIs. The tariff code being used can be found as an attribute on the current rate sensor. If you follow [these instructions](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq/#ive-been-asked-for-my-meter-information-in-a-bug-request-how-do-i-obtain-this), you can see the account information being provided by OE via their API. This will probably shed some light on why it's picking the wrong tariff.

**Reply by blacknell** [reply]
Now that's getting interesting. 

The attribute says the tariff being used is E-1R-VAR-22-11-01-H which is correct. 

However, the price is wrong. In the attribute it says 0.25 (unfortunately rounded) but I see 0.253578 in the sensor.

In my API call results above you can see it's 0.2461704

So something is happening here.

**Reply by BottlecapDave** [reply, marked as answer]
It might be direct debit vs non direct debit pricing. You can configure this in your main integration settings (https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/account/#favour-direct-debit-rates)

**Reply by blacknell** [reply]
Ok. That was it. I'm not sure what the default should be but I guess the user has to pay attention to this when configuring. Maybe it could go higher up on that dialog page. 

Maybe also an attribute stating yes or no for whether the tariff is DD. 

Thanks for helping.

**Reply by blacknell** [reply]
Also, since this was the case how come the tariff came back with correct values when I queried. I could see reference to DD

**Reply by BottlecapDave** [reply]
If you followed the link to the standard unit rates, which is what th integration uses, you'd have seen two entries for each day, one for direct debit and one for without. I'll have a look at adding a direct debit attribute to the rate sensors.
